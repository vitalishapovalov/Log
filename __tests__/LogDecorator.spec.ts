import {
    T,
    TConstructor,
    TGetPrototypeOf,
    TIsExtensible,
    TOwnKeys,
    TPreventExtensions,
    TSetPrototypeOf,
    TLogFullProperty,
    TFunction,
    TProvideLogger, TMethod
} from "./__mocks__/LogDecorator";

// TODO when used as a parameter decorator
// TODO when used as a method decorator
// TODO when used as a getset decorator
// TODO when used on a class + property/method/getset
// TODO static properties/methods/getset
// TODO cover different options cases

@Describe("log/Log test, when used as a class decorator")
class LogDecoratorSpec1 {

    dummy;

    @Spy(console, "log")
    logSpy;

    @It("shouldn't log anything")
    test1() {
        const inst = new T();

        this.dummy = Object.getPrototypeOf(inst);
        Object.isExtensible(inst);
        Object.setPrototypeOf(inst, Object.create({}));
        Object.preventExtensions(inst);

        expect(this.logSpy).not.toHaveBeenCalled();
    }

    @It("should correctly log only get/set events on property")
    test2() {
        const inst = new T();

        this.dummy = inst.prop;
        inst.prop = "";

        // shouldn't be logged
        this.dummy = "prop" in inst;
        delete inst.prop;
        Object.defineProperty(inst, "prop", { value: "prop_value" });
        this.dummy = Object.getOwnPropertyDescriptor(inst, "prop");

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(JSON.stringify(this.logSpy.mock.calls[1])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(2);
    }

    @It("should correctly log only get/set events on getters and setters")
    testThree() {
        const inst = new T();

        this.dummy = inst.getSetProp;
        inst.getSetProp = "";

        // shouldn't be logged
        this.dummy = "getSetProp" in inst;
        delete inst.getSetProp;
        Object.defineProperty(inst, "getSetProp", { value: "getSetProp_value" });
        this.dummy = Object.getOwnPropertyDescriptor(inst, "getSetProp");

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(JSON.stringify(this.logSpy.mock.calls[1])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(2);
    }

    @It("should correctly log only call events on methods")
    testFour() {
        const inst = new T();

        this.dummy = inst.method();

        // shouldn't be logged
        this.dummy = "method" in inst;
        delete inst.method;
        Object.defineProperty(inst, "method", { value: () => {} });
        this.dummy = Object.getOwnPropertyDescriptor(inst, "method");

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(1);
    }

    @It("should correctly log constructor")
    testFive() {
        new TConstructor({ a: 10 }, 100, "100", ["str"]);

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(1);
    }

    @It("should correctly log getPrototypeOf")
    testSix() {
        Object.getPrototypeOf(new TGetPrototypeOf());

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(1);
    }

    @It("should correctly log setPrototypeOf")
    testSeven() {
        Object.setPrototypeOf(new TSetPrototypeOf(), {
            dummyMethod() {}
        });

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(1);
    }

    @It("should correctly log isExtensible")
    testEight() {
        Object.isExtensible(new TIsExtensible());

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(1);
    }

    @It("should correctly log preventExtensions")
    testNine() {
        Object.preventExtensions(new TPreventExtensions());

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(1);
    }

    @It("should correctly log ownKeys")
    testTen() {
        const instance = new TOwnKeys();

        Object.getOwnPropertyNames(instance);
        Object.getOwnPropertySymbols(instance);
        Reflect.ownKeys(instance);
        Object.keys(instance);
        for (const k in instance) {}

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(JSON.stringify(this.logSpy.mock.calls[1])).toMatchSnapshot();
        expect(JSON.stringify(this.logSpy.mock.calls[2])).toMatchSnapshot();
        expect(JSON.stringify(this.logSpy.mock.calls[3])).toMatchSnapshot();
        expect(JSON.stringify(this.logSpy.mock.calls[4])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(5);
    }

    @It("should correctly log delete")
    testEleven() {
        delete new TLogFullProperty().prop;

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(1);
    }

    @It("should correctly log defineProperty")
    testTwelve() {
        Object.defineProperty(new TLogFullProperty(), "prop", { value: "new val" });

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(1);
    }

    @It("should correctly log has")
    testThirteen() {
        this.dummy = "prop" in new TLogFullProperty();

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(1);
    }

    @It("should correctly log getOwnPropertyDescriptor")
    testFourteen() {
        Object.getOwnPropertyDescriptor(new TLogFullProperty(), "prop");

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(1);
    }

    @It("should correctly log apply")
    testFifteen() {
        const instance = new TFunction("a", "b", "return a + b;");
        instance.call({}, 20, 30);

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(1);
    }

    @It("should provide instance logger")
    testSixteen() {
        expect((new TProvideLogger() as any).logger).toBeTruthy();
    }

    @It("should correctly log messages")
    testSeventeen() {
        const logger = (new TProvideLogger() as any).logger;

        logger.log("");
        logger.info("Message []");
        logger.warn("Number: $number(1000)");
        logger.error("String: $string(alex)");

        expect(this.logSpy).toHaveBeenCalledTimes(4);
    }

    @It("should correctly log methods usage")
    testMethodLogging() {
        const instance = new TMethod();

        instance.myMethod("foo");
        instance.myNamedMethod("foo", "bar");

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(JSON.stringify(this.logSpy.mock.calls[1])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(2);
    }
}
