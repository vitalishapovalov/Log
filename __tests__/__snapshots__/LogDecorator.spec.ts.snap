// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`log/Log test, when used as a class decorator should correctly log apply 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m         \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTFunction\\\\u001b[39m                                       \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall \\\\u001b[91mfunction\\\\u001b[39m\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[93mTFunction\\\\u001b[39m\\\\u001b[97m(\\\\u001b[39m\\\\u001b[97marg1\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mNumber\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[97marg2\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mNumber\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mNumber\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall context\\\\u001b[39m   \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97m{\\\\u001b[39m\\\\u001b[97m}\\\\u001b[39m                                              \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall with\\\\u001b[39m      \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97m(\\\\u001b[39m\\\\u001b[36m20\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[36m30\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m                                        \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall result\\\\u001b[39m    \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[36m50\\\\u001b[39m                                              \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log constructor 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m                  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTConstructor\\\\u001b[39m                      \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mConstruct \\\\u001b[91mnew\\\\u001b[39m \\\\u001b[97minstance\\\\u001b[97m\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mObject\\\\u001b[39m                            \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mConstruct arguments\\\\u001b[39m     \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97m(\\\\u001b[39m\\\\u001b[97m{\\\\u001b[39m \\\\u001b[35ma\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[36m10\\\\u001b[39m \\\\u001b[97m}\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[36m100\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[32m\\\\\\"100\\\\\\"\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[97m[\\\\u001b[39m\\\\u001b[32m\\\\\\"str\\\\\\"\\\\u001b[39m\\\\u001b[97m]\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m  \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log defineProperty 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m                  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTLogFullProperty\\\\u001b[39m      \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall \\\\u001b[93mdefineProperty\\\\u001b[97m \\\\u001b[97mon\\\\u001b[97m\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97mTLogFullProperty\\\\u001b[39m      \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mProperty key\\\\u001b[39m            \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[35mprop\\\\u001b[39m                  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mProperty descriptor\\\\u001b[39m     \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97m{\\\\u001b[39m \\\\u001b[35mvalue\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[32m\\\\\\"new val\\\\\\"\\\\u001b[39m \\\\u001b[97m}\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall result\\\\u001b[39m             \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mtrue\\\\u001b[39m                  \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log delete 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m                  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTLogFullProperty\\\\u001b[39m        \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mAccess \\\\u001b[91mdelete\\\\u001b[39m \\\\u001b[97moperator\\\\u001b[97m\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mdelete\\\\u001b[39m \\\\u001b[91mObject\\\\u001b[39m\\\\u001b[97m[\\\\u001b[39m\\\\u001b[32m\\\\\\"prop\\\\\\"\\\\u001b[39m\\\\u001b[97m]\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mDelete result\\\\u001b[39m           \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mtrue\\\\u001b[39m                    \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log getOwnPropertyDescriptor 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m                            \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTLogFullProperty\\\\u001b[39m                              \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall \\\\u001b[93mgetOwnPropertyDescriptor\\\\u001b[97m \\\\u001b[97mon\\\\u001b[97m\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[32m\\\\\\"prop\\\\\\"\\\\u001b[39m                                        \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mRetrieved descriptor\\\\u001b[39m              \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97m{\\\\u001b[39m \\\\u001b[35mvalue\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[32m\\\\\\"prop_value\\\\\\"\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[35mwritable\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[91mtrue\\\\u001b[39m\\\\u001b[97m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[97m.\\\\u001b[39m\\\\u001b[97m.\\\\u001b[39m\\\\u001b[97m.\\\\u001b[39m\\\\u001b[39m \\\\u001b[97m}\\\\u001b[39m  \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log getPrototypeOf 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m                  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTGetPrototypeOf\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall \\\\u001b[93mgetPrototypeOf\\\\u001b[97m \\\\u001b[97mof\\\\u001b[97m\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97mTGetPrototypeOf\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mRetrieved prototype\\\\u001b[39m     \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mObject\\\\u001b[39m           \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log has 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m              \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTLogFullProperty\\\\u001b[39m            \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mAccess \\\\u001b[91min\\\\u001b[39m \\\\u001b[97moperator\\\\u001b[97m\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[32m\\\\\\"prop\\\\\\"\\\\u001b[39m \\\\u001b[91min\\\\u001b[39m \\\\u001b[97mTLogFullProperty\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mAccess result\\\\u001b[39m       \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mtrue\\\\u001b[39m                        \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log isExtensible 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m                 \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTIsExtensible\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCheck \\\\u001b[93misExtensible\\\\u001b[97m \\\\u001b[97mof\\\\u001b[97m\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97mTIsExtensible\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mResult\\\\u001b[39m                 \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mtrue\\\\u001b[39m           \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log only call events on methods 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m       \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mT\\\\u001b[39m                  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall \\\\u001b[93mmethod\\\\u001b[97m\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[93mmethod\\\\u001b[39m\\\\u001b[97m(\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mString\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall result\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[32m\\\\\\"method_value\\\\\\"\\\\u001b[39m     \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log only get/set events on getters and setters 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m           \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mT\\\\u001b[39m                          \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mAccess property\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mget\\\\u001b[39m \\\\u001b[93mgetSetProp\\\\u001b[39m\\\\u001b[97m(\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mString\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mAccess result\\\\u001b[39m    \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[32m\\\\\\"getter_getSetProp\\\\\\"\\\\u001b[39m        \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log only get/set events on getters and setters 2`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m        \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mT\\\\u001b[39m                             \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mSet property\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mset\\\\u001b[39m \\\\u001b[93mgetSetProp\\\\u001b[39m\\\\u001b[97m(\\\\u001b[39m\\\\u001b[97marg\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mString\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mSet \\\\u001b[32msucceed\\\\u001b[39m\\\\u001b[39m   \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[93mgetSetProp\\\\u001b[39m \\\\u001b[97m=\\\\u001b[39m \\\\u001b[32m\\\\\\"\\\\\\"\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m              \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log only get/set events on property 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m           \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mT\\\\u001b[39m              \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mAccess property\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[35mprop\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mString\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mAccess result\\\\u001b[39m    \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[32m\\\\\\"prop_value\\\\\\"\\\\u001b[39m   \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log only get/set events on property 2`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m        \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mT\\\\u001b[39m              \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mSet property\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[35mprop\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mString\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mSet \\\\u001b[32msucceed\\\\u001b[39m\\\\u001b[39m   \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[35mprop\\\\u001b[39m \\\\u001b[97m=\\\\u001b[39m \\\\u001b[32m\\\\\\"\\\\\\"\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m     \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log ownKeys 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m           \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTOwnKeys\\\\u001b[39m                    \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mGet own keys of\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97mTOwnKeys\\\\u001b[39m                    \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mResult\\\\u001b[39m           \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97m[\\\\u001b[39m\\\\u001b[32m\\\\\\"key1\\\\\\"\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[32m\\\\\\"key2\\\\\\"\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[97mSymbol\\\\u001b[97m(\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m\\\\u001b[39m\\\\u001b[97m]\\\\u001b[39m  \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log ownKeys 2`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m           \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTOwnKeys\\\\u001b[39m                    \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mGet own keys of\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97mTOwnKeys\\\\u001b[39m                    \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mResult\\\\u001b[39m           \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97m[\\\\u001b[39m\\\\u001b[32m\\\\\\"key1\\\\\\"\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[32m\\\\\\"key2\\\\\\"\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[97mSymbol\\\\u001b[97m(\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m\\\\u001b[39m\\\\u001b[97m]\\\\u001b[39m  \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log ownKeys 3`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m           \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTOwnKeys\\\\u001b[39m                    \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mGet own keys of\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97mTOwnKeys\\\\u001b[39m                    \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mResult\\\\u001b[39m           \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97m[\\\\u001b[39m\\\\u001b[32m\\\\\\"key1\\\\\\"\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[32m\\\\\\"key2\\\\\\"\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[97mSymbol\\\\u001b[97m(\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m\\\\u001b[39m\\\\u001b[97m]\\\\u001b[39m  \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log ownKeys 4`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m           \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTOwnKeys\\\\u001b[39m                    \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mGet own keys of\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97mTOwnKeys\\\\u001b[39m                    \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mResult\\\\u001b[39m           \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97m[\\\\u001b[39m\\\\u001b[32m\\\\\\"key1\\\\\\"\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[32m\\\\\\"key2\\\\\\"\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[97mSymbol\\\\u001b[97m(\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m\\\\u001b[39m\\\\u001b[97m]\\\\u001b[39m  \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log ownKeys 5`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m           \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTOwnKeys\\\\u001b[39m                    \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mGet own keys of\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97mTOwnKeys\\\\u001b[39m                    \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mResult\\\\u001b[39m           \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97m[\\\\u001b[39m\\\\u001b[32m\\\\\\"key1\\\\\\"\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[32m\\\\\\"key2\\\\\\"\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[97mSymbol\\\\u001b[97m(\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m\\\\u001b[39m\\\\u001b[97m]\\\\u001b[39m  \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log preventExtensions 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m                     \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTPreventExtensions\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall \\\\u001b[93mpreventExtensions\\\\u001b[97m \\\\u001b[97mon\\\\u001b[97m\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97mTPreventExtensions\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mResult\\\\u001b[39m                     \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mtrue\\\\u001b[39m                \\\\u001b[49m\\"]"`;

exports[`log/Log test, when used as a class decorator should correctly log setPrototypeOf 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m                  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTSetPrototypeOf\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall \\\\u001b[93msetPrototypeOf\\\\u001b[97m \\\\u001b[97mon\\\\u001b[97m\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97mTSetPrototypeOf\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mSet \\\\u001b[32msucceed\\\\u001b[39m\\\\u001b[39m             \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mObject\\\\u001b[39m           \\\\u001b[49m\\"]"`;
